### ISSUE TITLE
Issue : VM IMAGE NOT FOUND when we run = vagrant up --provider=azure

### ISSUE THREAD

**Issue by arindam00** [MSOpenTech/vagrant-azure#147]
Below is the VAGRANTFILE I am using and when I run the command "vagrant up --provider=azure"
=======================================================
Vagrant.configure('2') do |config|
  config.vm.box = 'azure'

  # use local ssh key to connect to remote vagrant box
  config.ssh.private_key_path = 'C:\Users\ardhar\Downloads\privatekey.ppk'
  config.vm.provider :azure do |azure, override|

    # use Azure Active Directory Application / Service Principal to connect to Azure
    # see: https://azure.microsoft.com/en-us/documentation/articles/resource-group-create-service-principal-portal/

    # each of the below values will default to use the env vars named as below if not specified explicitly
    azure.tenant_id = '******************************************'
    azure.client_id = '*******************************************'
    azure.client_secret = '***************************************'
    azure.subscription_id = '**************************************'
  end

end
=======================================================
when I run the command "vagrant up --provider=azure" I get the below error 
========================================================
C:\Users\ardhar\.vagrant.d\boxes\azure\0\azure>vagrant up --provider=azure
Bringing machine 'default' up with 'azure' provider...
==> default: Preparing SMB shared folders...
    default: You will be asked for the username and password to use for the SMB
    default: folders shortly. Please use the proper username/password of your
    default: Windows account.
    default:
    default: Username: arindam
    default: Password (will be hidden):
==> default: Launching an instance with the following settings...
==> default:  -- Management Endpoint: https://management.azure.com
==> default:  -- Subscription Id: *****************************************
==> default:  -- Resource Group Name: red-darkness-24
==> default:  -- Location: westus
==> default:  -- VM Name: blue-bird-18
==> default:  -- VM Size: Standard_D1
==> default:  -- Image URN: canonical:ubuntuserver:16.04.0-DAILY-LTS:latest
C:/Users/ardhar/.vagrant.d/gems/2.2.5/gems/azure_mgmt_compute-0.2.1/lib/azure_mgmt_compute/virtual_machine_images.rb:139:in `block in list': { (MsRestAzure::AzureOperationError)
  "message": "MsRestAzure::AzureOperationError",
  "request": {
    "base_uri": "https://management.azure.com",
    "path_template": "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/publishers/{publisherName}/artifacttypes/vmimage/offers/{offer}/skus/{skus}/versions",
    "method": "get",
    "path_params": {
      "location": "westus",
      "publisherName": "canonical",
      "offer": "ubuntuserver",
      "skus": "16.04.0-DAILY-LTS",
      "subscriptionId": "**********************************"
    },
    "skip_encoding_path_params": null,
    "query_params": {
      "$filter": null,
      "$top": null,
      "$orderby": null,
      "api-version": "2015-06-15"
    },
    "skip_encoding_query_params": null,
    "headers": {
      "x-ms-client-request-id": "e6c3bb49-4761-4f7a-9727-5bf47348439b",
      "accept-language": "en-US"
    },
    "body": null,
    "middlewares": [
      [
        "MsRest::RetryPolicyMiddleware",
        {
          "times": 3,
          "retry": 0.02
        }
      ],
      [
        "cookie_jar"
      ]
    ],
    "log": null
  },
  "response": {
    "body": "{\r\n  \"error\": {\r\n    \"code\": \"NotFound\",\r\n    \"message\": \"Artifact: VMImage was not found.\"\r\n  }\r\n}",
    "headers": {
      "cache-control": "no-cache",
      "pragma": "no-cache",
      "content-length": "99",
      "content-type": "application/json; charset=utf-8",
      "expires": "-1",
      "strict-transport-security": "max-age=31536000; includeSubDomains",
      "x-ms-request-id": "0d5892eb-99ca-4741-a906-f71f6442fbbc",
      "server": "Microsoft-HTTPAPI/2.0, Microsoft-HTTPAPI/2.0",
      "x-ms-ratelimit-remaining-subscription-reads": "14699",
      "x-ms-correlation-request-id": "3bff9ae9-1934-4765-ba11-7adcd21a7e0b",
      "x-ms-routing-request-id": "SOUTHINDIA:20161228T161854Z:3bff9ae9-1934-4765-ba11-7adcd21a7e0b",
      "date": "Wed, 28 Dec 2016 16:18:54 GMT",
      "connection": "close"
    },
    "status": 404
  }
}
        from C:/Users/ardhar/.vagrant.d/gems/2.2.5/gems/concurrent-ruby-1.0.4/lib/concurrent/promise.rb:501:in `call'
        from C:/Users/ardhar/.vagrant.d/gems/2.2.5/gems/concurrent-ruby-1.0.4/lib/concurrent/promise.rb:501:in `block in on_fulfill'
        from C:/Users/ardhar/.vagrant.d/gems/2.2.5/gems/concurrent-ruby-1.0.4/lib/concurrent/executor/safe_task_executor.rb:24:in `call'
        from C:/Users/ardhar/.vagrant.d/gems/2.2.5/gems/concurrent-ruby-1.0.4/lib/concurrent/executor/safe_task_executor.rb:24:in `block in execute'
        from C:/Users/ardhar/.vagrant.d/gems/2.2.5/gems/concurrent-ruby-1.0.4/lib/concurrent/synchronization/mri_lockable_object.rb:38:in `block in synchronize'
        from C:/Users/ardhar/.vagrant.d/gems/2.2.5/gems/concurrent-ruby-1.0.4/lib/concurrent/synchronization/mri_lockable_object.rb:38:in `synchronize'
        from C:/Users/ardhar/.vagrant.d/gems/2.2.5/gems/concurrent-ruby-1.0.4/lib/concurrent/synchronization/mri_lockable_object.rb:38:in `synchronize'
        from C:/Users/ardhar/.vagrant.d/gems/2.2.5/gems/concurrent-ruby-1.0.4/lib/concurrent/executor/safe_task_executor.rb:19:in `execute'
        from C:/Users/ardhar/.vagrant.d/gems/2.2.5/gems/concurrent-ruby-1.0.4/lib/concurrent/promise.rb:531:in `block in realize'
        from C:/Users/ardhar/.vagrant.d/gems/2.2.5/gems/concurrent-ruby-1.0.4/lib/concurrent/executor/ruby_thread_pool_executor.rb:348:in `call'
        from C:/Users/ardhar/.vagrant.d/gems/2.2.5/gems/concurrent-ruby-1.0.4/lib/concurrent/executor/ruby_thread_pool_executor.rb:348:in `run_task'
        from C:/Users/ardhar/.vagrant.d/gems/2.2.5/gems/concurrent-ruby-1.0.4/lib/concurrent/executor/ruby_thread_pool_executor.rb:337:in `block (3 levels) in create_worker'
        from C:/Users/ardhar/.vagrant.d/gems/2.2.5/gems/concurrent-ruby-1.0.4/lib/concurrent/executor/ruby_thread_pool_executor.rb:320:in `loop'
        from C:/Users/ardhar/.vagrant.d/gems/2.2.5/gems/concurrent-ruby-1.0.4/lib/concurrent/executor/ruby_thread_pool_executor.rb:320:in `block (2 levels) in create_worker'
        from C:/Users/ardhar/.vagrant.d/gems/2.2.5/gems/concurrent-ruby-1.0.4/lib/concurrent/executor/ruby_thread_pool_executor.rb:319:in `catch'
        from C:/Users/ardhar/.vagrant.d/gems/2.2.5/gems/concurrent-ruby-1.0.4/lib/concurrent/executor/ruby_thread_pool_executor.rb:319:in `block in create_worker'
============================================================
Upon Investigation I  found that "Image URN: canonical:ubuntuserver:16.04.0-DAILY-LTS:latest" is a Classic Image but the deployment is ARM as it is calling "Microsoft.Compute" 
=======
""path_template": "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/publishers/{publisherName}/artifacttypes/vmimage/offers/{offer}/skus/{skus}/versions","
=======

>> Can you please help in fixing this issue. Appreciate your help Thanks

**Comment by imonthercks** [comment]
I am seeing the same exception trying to run on Windows 10 with a specific location and vm_image_urn or if I just leave as defaults.

Here is my VagrantFile

```ruby
# Vagrantfile API/syntax version. Don't touch unless you know what you're doing!
VAGRANTFILE_API_VERSION = '2'

Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|

  config.vm.box = 'azure'
  
  config.vm.provider :azure do |azure, override|
	override.vm.synced_folder ".", "/vagrant", disabled: true
    azure.tenant_id = ENV['ARM_TENANT_ID']
    azure.client_id = ENV['ARM_CLIENT_ID']
    azure.client_secret = ENV['ARM_CLIENT_SECRET']
    azure.subscription_id = ENV['ARM_SUBSCRIPTION_ID']
	
	azure.location = 'eastus'
	azure.vm_image_urn = 'MicrosoftVisualStudio:VisualStudio:VS-2015-Comm-AzureSDK-2.9-WS2012R2:50.0.14'
  end
end
```
response from Azure API is:

 "response": {
    "body": "{\r\n  \"error\": {\r\n    \"code\": \"NotFound\",\r\n    \"message\": \"Artifact: VMImage was not found.\"\r\n  }\r\n}"

Just curious if I am missing something obvious before I spend a lot of time trying to dig further.

**Comment by AlexZeitler** [comment]
Same here - did you find a solution?

**Comment by cecchisandrone** [comment]
For example, I specified OpenLogic:CentOS:6.5:6.5.20160309 as azure.vm_image_urn and it seems to work.

**Comment by AlexZeitler** [comment]
I tried `MicrosoftWindowsServer:WindowsServer:2016-Datacenter-with-Containers:2016.0.20161213` now but then I get `The value for the template parameter 'adminPassword' at line '1' and column '306' is not provided. Please see https://aka.ms/arm-deploy/#parameter-file for usage details.`

**Comment by devigned** [comment]
Should be fixed with the 2.0.0.pre2 release. Please reopen if not the case.

**Comment by beaudryj** [comment]
This is looking to be a problem in pre8 :/
